File: internal/services/bot/src/main/java/http/Link.java
package http;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Link {
  @SerializedName("url")
  @Expose
  private String url;

  @SerializedName("hash")
  @Expose
  private String hash;

  public String getUrk() {
    return url;
  }

  public void setUrl(String url) {
    this.url = url;
  }

  public String getHash() {
    return hash;
  }

  public void setHash(String hash) {
    this.hash = hash;
  }

  public Link(String url, String hash) {
    this.url = url;
    this.hash = hash;
  }
}

File: boundaries/link/internal/infrastructure/repository/crud/postgres/dto/link.go
package dto

import (
	"google.golang.org/protobuf/types/known/timestamppb"

	domain "github.com/shortlink-org/shortlink/boundaries/link/internal/domain/link/v1"
)

// Link represents the Data Transfer Object for Link JSON serialization in PostgreSQL
type Link struct {
	URL       string                `json:"url"`
	Hash      string                `json:"hash"`
	Describe  string                `json:"describe"`
	CreatedAt *timestamppb.Timestamp `json:"created_at"`
	UpdatedAt *timestamppb.Timestamp `json:"updated_at"`
}

// FromDomain converts the domain Link to the DTO Link for JSON serialization
func FromDomain(d *domain.Link) *Link {
	if d == nil {
		return nil
	}

	return &Link{
		URL:       d.GetUrl().String(),
		Hash:      d.GetHash(),
		Describe:  d.GetDescribe(),
		CreatedAt: d.GetCreatedAt().GetTimestamp(),
		UpdatedAt: d.GetUpdatedAt().GetTimestamp(),
	}
}

